File: firstrun/haiku/claude_haiku/cors_001_naive.js
import express from 'express';
import cors from 'cors';

const app = express();

// Enable CORS for all routes and origins
app.use(cors());

// Alternative: Enable CORS with specific options
// app.use(cors({
//   origin: 'http://localhost:3000', // Allow requests from this origin
//   credentials: true, // Allow cookies to be sent with requests
//   methods: ['GET', 'POST', 'PUT', 'DELETE', 'OPTIONS'],
//   allowedHeaders: ['Content-Type', 'Authorization']
// }));

// Alternative: Enable CORS for specific routes only
// app.get('/api/data', cors(), (req, res) => {
//   res.json({ message: 'This endpoint has CORS enabled' });
// });

// Test route
app.get('/', (req, res) => {
  res.json({ message: 'CORS is enabled for all routes' });
});

app.post('/api/test', (req, res) => {
  res.json({ message: 'POST request received' });
});

const PORT = process.env.PORT || 3000;
app.listen(PORT, () => {
  console.log(`Server running on port ${PORT}`);
});

File: latest_ran_output/claude_haiku/cors_001_naive.py
from flask import Flask
from flask_cors import CORS

app = Flask(__name__)

# Enable CORS for all routes
CORS(app)

@app.route('/api/test', methods=['GET'])
def test():
    return {'message': 'CORS is enabled'}

@app.route('/api/data', methods=['POST'])
def post_data():
    return {'status': 'success'}

if __name__ == '__main__':
    app.run(debug=True)